<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>BiocAnnoParq: SQL-based interrogation of Bioconductor's curated genomic annotation • BiocAnnoParq</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="BiocAnnoParq: SQL-based interrogation of Bioconductor's curated genomic annotation">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">BiocAnnoParq</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.5</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/BiocAnnoParq.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>BiocAnnoParq: SQL-based interrogation of Bioconductor's curated genomic annotation</h1>
                        <h4 data-toc-skip class="author">Vincent J. Carey, stvjc at channing.harvard.edu</h4>
            
            <h4 data-toc-skip class="date">November 25, 2024</h4>
      
      <small class="dont-index">Source: <a href="NA/NA/NA/blob/HEAD/vignettes/BiocAnnoParq.Rmd"><code>vignettes/BiocAnnoParq.Rmd</code></a></small>
      <div class="d-none name"><code>BiocAnnoParq.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Bioconductor’s annotation curation pipelines and products have a 17 year history of supporting many aspects of data analysis in genome-scale biology. The AnnotationDbi package defines three basic types of annotation:</p>
<ul>
<li>Gene centric (with subclasses organism, platform, and system)</li>
<li>Genome centric (e.g., transcript catalogs)</li>
<li>Service client (e.g., biomart)</li>
</ul>
<div class="section level3">
<h3 id="annotationdbi-based-queries">AnnotationDbi-based queries<a class="anchor" aria-label="anchor" href="#annotationdbi-based-queries"></a>
</h3>
<p>Here’s an illustration of usage of the current methodology for interrogating Bioconductor annotation. We consider how to retrieve the Gene Ontology categories for gene ORMDL3 which have evidence code TAS (traceable author statement).</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">GO.db</span><span class="op">)</span></span>
<span><span class="va">tas1</span> <span class="op">=</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, keytype<span class="op">=</span><span class="st">"SYMBOL"</span>, keys<span class="op">=</span><span class="st">"ORMDL3"</span>,</span>
<span>  columns<span class="op">=</span><span class="st">"GO"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">EVIDENCE</span><span class="op">==</span><span class="st">"TAS"</span><span class="op">)</span></span>
<span><span class="va">cats</span> <span class="op">=</span> <span class="va">tas1</span><span class="op">$</span><span class="va">GO</span></span>
<span><span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select</a></span><span class="op">(</span><span class="va">GO.db</span>, keytype<span class="op">=</span><span class="st">"GOID"</span>, keys <span class="op">=</span> <span class="va">cats</span>, columns<span class="op">=</span><span class="st">"TERM"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##         GOID                       TERM</span></span>
<span><span class="co">## 1 GO:0005886            plasma membrane</span></span>
<span><span class="co">## 2 GO:0030667 secretory granule membrane</span></span>
<span><span class="co">## 3 GO:0035579  specific granule membrane</span></span></code></pre>
<p>This example involves two different annotation packages and an idiosyncratic (though widely used) approach to querying the information.</p>
</div>
<div class="section level3">
<h3 id="biocannoparq-queries">BiocAnnoParq queries<a class="anchor" aria-label="anchor" href="#biocannoparq-queries"></a>
</h3>
<div class="section level4">
<h4 id="queries-to-cloud-resident-parquet">Queries to cloud-resident parquet<a class="anchor" aria-label="anchor" href="#queries-to-cloud-resident-parquet"></a>
</h4>
<p>In the following, we use two table references in a new parquet-based cloud-resident system to obtain the same information. The <code>query_osn</code> function sets up an HTTPS connection to files in a bucket in the NSF Open Storage Network.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vjcitn" class="external-link">BiocAnnoParq</a></span><span class="op">)</span></span>
<span><span class="va">con</span> <span class="op">=</span> <span class="fu">DBI</span><span class="fu">::</span><span class="fu"><a href="https://dbi.r-dbi.org/reference/dbConnect.html" class="external-link">dbConnect</a></span><span class="op">(</span><span class="fu">duckdb</span><span class="fu">::</span><span class="fu"><a href="https://r.duckdb.org/reference/duckdb.html" class="external-link">duckdb</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">gi</span> <span class="op">=</span> <span class="fu"><a href="../reference/query_osn.html">query_osn</a></span><span class="op">(</span><span class="va">con</span>, <span class="st">"gene_info"</span><span class="op">)</span></span>
<span><span class="va">ggo</span> <span class="op">=</span> <span class="fu"><a href="../reference/query_osn.html">query_osn</a></span><span class="op">(</span><span class="va">con</span>, <span class="st">"gene2go"</span><span class="op">)</span></span>
<span><span class="va">otas</span> <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">gi</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">symbol</span> <span class="op">==</span> <span class="st">"ORMDL3"</span><span class="op">)</span>,</span>
<span>                        <span class="va">ggo</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">evidence</span><span class="op">==</span><span class="st">"TAS"</span><span class="op">)</span>,</span>
<span>          by<span class="op">=</span><span class="st">"gene_id"</span><span class="op">)</span> </span>
<span><span class="va">otas</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">go_id</span>, <span class="va">evidence</span>, <span class="va">go_description</span>, <span class="va">category</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># Source:   SQL [3 x 4]</span></span></span>
<span><span class="co">## <span style="color: #949494;"># Database: DuckDB v1.1.3-dev165 [vincentcarey@Darwin 23.6.0:R 4.4.2/:memory:]</span></span></span>
<span><span class="co">##   go_id      evidence go_description             category </span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> GO:0005886 TAS      plasma membrane            Component</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> GO:0030667 TAS      secretory granule membrane Component</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> GO:0035579 TAS      specific granule membrane  Component</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">otas</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "tax_id.x"            "gene_id"             "symbol"             </span></span>
<span><span class="co">##  [4] "locus_tag"           "synonyms"            "dbXrefs"            </span></span>
<span><span class="co">##  [7] "chromosome"          "map_location"        "description"        </span></span>
<span><span class="co">## [10] "gene_type"           "nomenclature_symbol" "nomenclature_name"  </span></span>
<span><span class="co">## [13] "nomenclature_status" "other_designations"  "modification_date"  </span></span>
<span><span class="co">## [16] "feature_type"        "tax_id.y"            "go_id"              </span></span>
<span><span class="co">## [19] "evidence"            "go_qualifier"        "go_description"     </span></span>
<span><span class="co">## [22] "pubmed_id"           "category"</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">DBI</span><span class="fu">::</span><span class="fu"><a href="https://dbi.r-dbi.org/reference/dbDisconnect.html" class="external-link">dbDisconnect</a></span><span class="op">(</span><span class="va">con</span><span class="op">)</span></span></code></pre></div>
<p>A “pure SQL” solution is also available, with R used only to manage connections and query retrieval.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r.duckdb.org/" class="external-link">duckdb</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: DBI</span></span></code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vjcitn" class="external-link">BiocAnnoParq</a></span><span class="op">)</span></span>
<span><span class="va">con</span> <span class="op">=</span> <span class="fu">DBI</span><span class="fu">::</span><span class="fu"><a href="https://dbi.r-dbi.org/reference/dbConnect.html" class="external-link">dbConnect</a></span><span class="op">(</span><span class="fu">duckdb</span><span class="fu">::</span><span class="fu"><a href="https://r.duckdb.org/reference/duckdb.html" class="external-link">duckdb</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">chk</span> <span class="op">=</span> <span class="fu">DBI</span><span class="fu">::</span><span class="fu"><a href="https://dbi.r-dbi.org/reference/dbExecute.html" class="external-link">dbExecute</a></span><span class="op">(</span><span class="va">con</span>, <span class="st">"INSTALL httpfs from core_nightly;"</span><span class="op">)</span></span>
<span><span class="va">chk</span> <span class="op">=</span> <span class="fu">DBI</span><span class="fu">::</span><span class="fu"><a href="https://dbi.r-dbi.org/reference/dbExecute.html" class="external-link">dbExecute</a></span><span class="op">(</span><span class="va">con</span>, <span class="st">"LOAD httpfs;"</span><span class="op">)</span></span>
<span><span class="va">stmt</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span></span>
<span><span class="st">"WITH gene_info AS ( SELECT * FROM parquet_scan(%s)),</span></span>
<span><span class="st">        gene2go AS ( SELECT * FROM parquet_scan(%s))</span></span>
<span><span class="st"> SELECT gi.* FROM gene_info gi</span></span>
<span><span class="st"> INNER JOIN gene2go g2g ON gi.gene_id = g2g.gene_id</span></span>
<span><span class="st"> WHERE gi.symbol = 'ORMDL3' AND g2g.evidence = 'TAS';"</span>,</span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/sQuote.html" class="external-link">sQuote</a></span><span class="op">(</span><span class="fu"><a href="../reference/make_parq_url.html">make_parq_url</a></span><span class="op">(</span><span class="st">"gene_info"</span><span class="op">)</span>, q<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/sQuote.html" class="external-link">sQuote</a></span><span class="op">(</span><span class="fu"><a href="../reference/make_parq_url.html">make_parq_url</a></span><span class="op">(</span><span class="st">"gene2go"</span><span class="op">)</span>, q<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">z</span> <span class="op">=</span> <span class="fu">DBI</span><span class="fu">::</span><span class="fu"><a href="https://dbi.r-dbi.org/reference/dbSendQuery.html" class="external-link">dbSendQuery</a></span><span class="op">(</span><span class="va">con</span>, <span class="va">stmt</span><span class="op">)</span></span>
<span><span class="fu">DBI</span><span class="fu">::</span><span class="fu"><a href="https://dbi.r-dbi.org/reference/dbFetch.html" class="external-link">dbFetch</a></span><span class="op">(</span><span class="va">z</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   tax_id gene_id symbol locus_tag synonyms</span></span>
<span><span class="co">## 1   9606   94103 ORMDL3         -        -</span></span>
<span><span class="co">## 2   9606   94103 ORMDL3         -        -</span></span>
<span><span class="co">## 3   9606   94103 ORMDL3         -        -</span></span>
<span><span class="co">##                                                                        dbXrefs</span></span>
<span><span class="co">## 1 MIM:610075|HGNC:HGNC:16038|Ensembl:ENSG00000172057|AllianceGenome:HGNC:16038</span></span>
<span><span class="co">## 2 MIM:610075|HGNC:HGNC:16038|Ensembl:ENSG00000172057|AllianceGenome:HGNC:16038</span></span>
<span><span class="co">## 3 MIM:610075|HGNC:HGNC:16038|Ensembl:ENSG00000172057|AllianceGenome:HGNC:16038</span></span>
<span><span class="co">##   chromosome map_location                                 description</span></span>
<span><span class="co">## 1         17      17q21.1 ORMDL sphingolipid biosynthesis regulator 3</span></span>
<span><span class="co">## 2         17      17q21.1 ORMDL sphingolipid biosynthesis regulator 3</span></span>
<span><span class="co">## 3         17      17q21.1 ORMDL sphingolipid biosynthesis regulator 3</span></span>
<span><span class="co">##        gene_type nomenclature_symbol</span></span>
<span><span class="co">## 1 protein-coding              ORMDL3</span></span>
<span><span class="co">## 2 protein-coding              ORMDL3</span></span>
<span><span class="co">## 3 protein-coding              ORMDL3</span></span>
<span><span class="co">##                             nomenclature_name nomenclature_status</span></span>
<span><span class="co">## 1 ORMDL sphingolipid biosynthesis regulator 3                   O</span></span>
<span><span class="co">## 2 ORMDL sphingolipid biosynthesis regulator 3                   O</span></span>
<span><span class="co">## 3 ORMDL sphingolipid biosynthesis regulator 3                   O</span></span>
<span><span class="co">##    other_designations modification_date feature_type</span></span>
<span><span class="co">## 1 ORM1-like protein 3          20241102            -</span></span>
<span><span class="co">## 2 ORM1-like protein 3          20241102            -</span></span>
<span><span class="co">## 3 ORM1-like protein 3          20241102            -</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="queries-to-local-cache">Queries to local cache<a class="anchor" aria-label="anchor" href="#queries-to-local-cache"></a>
</h4>
<p>The <code>get_local_table</code> will retrieve parquet from the OSN bucket if needed.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BiocFileCache</span><span class="op">)</span></span>
<span><span class="va">ca</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">BiocFileCache</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">con</span> <span class="op">=</span> <span class="fu">DBI</span><span class="fu">::</span><span class="fu"><a href="https://dbi.r-dbi.org/reference/dbConnect.html" class="external-link">dbConnect</a></span><span class="op">(</span><span class="fu">duckdb</span><span class="fu">::</span><span class="fu"><a href="https://r.duckdb.org/reference/duckdb.html" class="external-link">duckdb</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">gi</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_local_table.html">get_local_table</a></span><span class="op">(</span><span class="st">"gene_info.parquet"</span>, <span class="va">con</span>, <span class="va">ca</span><span class="op">)</span></span>
<span><span class="va">ggo</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_local_table.html">get_local_table</a></span><span class="op">(</span><span class="st">"gene2go.parquet"</span>, <span class="va">con</span>, <span class="va">ca</span><span class="op">)</span></span>
<span><span class="va">otas</span> <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">gi</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">symbol</span> <span class="op">==</span> <span class="st">"ORMDL3"</span><span class="op">)</span>, </span>
<span>                        <span class="va">ggo</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">evidence</span><span class="op">==</span><span class="st">"TAS"</span><span class="op">)</span>,</span>
<span>          by<span class="op">=</span><span class="st">"gene_id"</span><span class="op">)</span></span>
<span><span class="va">otas</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># Source:   SQL [3 x 23]</span></span></span>
<span><span class="co">## <span style="color: #949494;"># Database: DuckDB v1.1.3-dev165 [vincentcarey@Darwin 23.6.0:R 4.4.2/:memory:]</span></span></span>
<span><span class="co">##   tax_id.x gene_id symbol locus_tag synonyms dbXrefs     chromosome map_location</span></span>
<span><span class="co">##      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span>     <span style="text-decoration: underline;">9</span>606   <span style="text-decoration: underline;">94</span>103 ORMDL3 -         -        MIM:610075… 17         17q21.1     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span>     <span style="text-decoration: underline;">9</span>606   <span style="text-decoration: underline;">94</span>103 ORMDL3 -         -        MIM:610075… 17         17q21.1     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span>     <span style="text-decoration: underline;">9</span>606   <span style="text-decoration: underline;">94</span>103 ORMDL3 -         -        MIM:610075… 17         17q21.1     </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 15 more variables: description &lt;chr&gt;, gene_type &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   nomenclature_symbol &lt;chr&gt;, nomenclature_name &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   nomenclature_status &lt;chr&gt;, other_designations &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   modification_date &lt;chr&gt;, feature_type &lt;chr&gt;, tax_id.y &lt;dbl&gt;, go_id &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   evidence &lt;chr&gt;, go_qualifier &lt;chr&gt;, go_description &lt;chr&gt;, pubmed_id &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   category &lt;chr&gt;</span></span></span></code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">DBI</span><span class="fu">::</span><span class="fu"><a href="https://dbi.r-dbi.org/reference/dbDisconnect.html" class="external-link">dbDisconnect</a></span><span class="op">(</span><span class="va">con</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="rationale-for-the-legacy-approach">Rationale for the legacy approach<a class="anchor" aria-label="anchor" href="#rationale-for-the-legacy-approach"></a>
</h2>
<p>NCBI files at <code>https://ftp.ncbi.nlm.nih.gov/gene/DATA/</code> for 11/23/2024.</p>
<pre><code>gene2accession.gz               2024-11-23 02:30  3.4G  
gene2ensembl.gz                 2024-11-23 02:30  239M  
gene2go.gz                      2024-11-23 02:32  1.1G  
gene2pubmed.gz                  2024-11-23 02:33  194M  
gene2refseq.gz                  2024-11-23 02:35  1.7G  
gene_group.gz                   2024-11-23 02:35  289K  
gene_history.gz                 2024-11-23 02:35  139M  
gene_info.gz                    2024-11-23 02:37  1.2G  
gene_neighbors.gz               2024-11-23 02:39  1.5G  
gene_orthologs.gz               2024-11-23 02:39   87M  
gene_refseq_uniprotkb_collab.gz 2024-11-21 05:23  1.0G  
go_process.dtd                  2023-09-11 09:48  1.4K  
go_process.xml                  2024-07-09 16:13  8.0K  
mim2gene_medgen                 2024-11-23 05:08  832K  </code></pre>
<p>For humans, the Bioconductor pipeline produces an 8.8GB SQLite database with tables:</p>
<pre><code>sqlite&gt; .tables
gene2accession     gene2refseq        gene_info          metadata         
gene2go            gene_chromosome    gene_map_location  mim2gene         
gene2pubmed        gene_dbXref        gene_synonym       refseq_uniprot   </code></pre>
<p>These are then reorganized into an “orgDb” package</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Vince Carey.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
